pyroclast/common/adversarial.py

```diff
@@ -0,0 +1,82 @@
+import tensorflow as tf
+import numpy as np
+
+
+def fast_gradient_sign_method(feature_fn, classify_fn, x, y, eps, norm):
+    """
+    Args:
+        feature_fn (x -> z): where x has shape [NHWCh]
+        classify_fn (z -> y): where y has shape [NCl] and z identical to feature_fn
+        x (Tensor): data, assumed to be [NHWCh]
+        y (Tensor): labels, assumed int typed
+        eps (float): norm constraint on the delta
+        norm (1, 2, or np.inf): norm choosing constraint set to optimize
+
+    Returns:
+        perturbed_data (Tensor): adversarially perturbed data to minimize \
```

Comment:
Do we want to be returning the perturbed data or the perturbation itself? We might want to analyze the perturbation, so I think we should return the perturbation instead of the modified data.